deviceadvisor/script/DATestRun.py

```diff
@@ -0,0 +1,224 @@
+
+except:
+    client.delete_thing(thingName = thing_name)
+    print("[Device Advisor]Error: Failed to create certificate.")
+    exit(-1)
+
+##############################################
+# attach certification to thing
+try:
+    print("[Device Advisor]Info: Attach certificate to test thing...")
+    # attache the certificate to thing
+    client.attach_thing_principal(
+        thingName = thing_name,
+        principal = create_cert_response['certificateArn']
+    )
+
+    certificate_arn = create_cert_response['certificateArn']
+    certificate_id = create_cert_response['certificateId']
+
+except:
+    delete_thing_with_certi(thing_name, certificate_id ,certificate_arn )
+    print("[Device Advisor]Error: Failed to attach certificate.")
+    exit(-1)
+
+
+##############################################
+# install iot-device-sdk-v2
+subprocess.run("npm install --unsafe-perm", shell = True)
+
+##############################################
+# Run device advisor
+for test_name in DATestConfig['tests']:
+    try:
+        ######################################
+        # set default shadow, for shadow update, if the
+        # shadow does not exists, update will fail
+        payload_shadow = json.dumps(
+        {
+        "state": {
+            "desired": {
+                shadowProperty: shadowDefault
+                },
+            "reported": {
+                shadowProperty: shadowDefault
+                }
+            }
+        })
+        shadow_response = dataClient.update_thing_shadow(
+            thingName = thing_name,
+            payload = payload_shadow)
+        get_shadow_response = dataClient.get_thing_shadow(thingName = thing_name)
+        # make sure shadow is created before we go to next step
+        while(get_shadow_response is None): 
+            get_shadow_response = dataClient.get_thing_shadow(thingName = thing_name)
+        
+        # start device advisor test
+        # test_start_response
+        # {
+        # 'suiteRunId': 'string',
+        # 'suiteRunArn': 'string',
+        # 'createdAt': datetime(2015, 1, 1)
+        # }
+        print("[Device Advisor]Info: Start device advisor test: " + test_name)
+        test_start_response = deviceAdvisor.start_suite_run(
+        suiteDefinitionId=DATestConfig['test_suite_ids'][test_name],
+        suiteRunConfiguration={
+            'primaryDevice': {
+                'thingArn': create_thing_response['thingArn'],
+            },
+            'parallelRun': True
+        })
+
+        # get DA endpoint
+        endpoint_response = deviceAdvisor.get_endpoint(
+            thingArn = create_thing_response['thingArn']
+        )
+        os.environ['DA_ENDPOINT'] = endpoint_response['endpoint']
+
+        working_dir = os.getcwd()
+        exe_path = os.path.join("deviceadvisor/tests",DATestConfig['test_exe_path'][test_name])
+        os.chdir(exe_path)
+        subprocess.run("npm install --unsafe-perm", shell = True)
+
+        while True:
```

Comment:
What is the test timeout?  I assume the DA test will go into a failed state eventually, I'm just curious how long it will take.

Reply:
Device Advisor sets a 5 minutes time out for each test.